test/codegen/bin_expr.c

```diff
@@ -7,5 +7,9 @@ int main() {
   __builtin_print(-1 << 31);
   __builtin_print(65536 >> 16);
   __builtin_print(-1 >> 16);
+  __builtin_print(1 && 1);
+  __builtin_print(0 && 1);
+  __builtin_print(1 || 0);
+  __builtin_print(0 || 0);
```

Comment:
How about representing `true` with a value other than `1`, so that if we accidentally only treat `1` as `true`, this test will fail?